Framework/src/TaskRunner.cxx

```diff
@@ -421,16 +422,32 @@ void TaskRunner::finishCycle(DataAllocator& outputs)
   }
 }
 
+void TaskRunner::updateMonitoringStats(ProcessingContext& pCtx)
+{
+  mNumberMessagesReceivedInCycle++;
+  for (auto&& input : pCtx.inputs()) {
+    if (input.header != nullptr) {
+      const auto* header = o2::header::get<header::DataHeader*>(input.header);
```

Comment:
i am not sure if DataHeader is guaranteed for all non-empty messages. could you add a check for that? if there is no data header, a warning probably should be enough.